drivers/gles3/shaders/scene.glsl

```diff
@@ -383,21 +387,27 @@ float get_omni_spot_attenuation(float distance, float inv_range, float decay) {
 }
 
 #if !defined(DISABLE_LIGHT_OMNI) || (defined(ADDITIVE_OMNI) && defined(USE_ADDITIVE_LIGHTING))
-void light_process_omni(uint idx, vec3 vertex, vec3 eye_vec, vec3 normal, float roughness,
+void light_process_omni_vertex(uint idx, vec3 vertex, vec3 eye_vec, vec3 normal, float roughness,
 		inout vec3 diffuse_light, inout vec3 specular_light) {
 	vec3 light_rel_vec = omni_lights[idx].position - vertex;
 	float light_length = length(light_rel_vec);
 	float omni_attenuation = get_omni_spot_attenuation(light_length, omni_lights[idx].inv_radius, omni_lights[idx].attenuation);
 	vec3 color = omni_lights[idx].color * omni_attenuation; // No light shaders here, so combine.
 
-	light_compute(normal, normalize(light_rel_vec), eye_vec, color, false, roughness,
-			diffuse_light,
-			specular_light);
+	// Compute area light solid angle.
+	float size_A = 0.0;
+	if (omni_lights[idx].size > 0.0) {
+		float t = omni_lights[idx].size / max(0.001, light_length);
+		size_A = max(0.0, 1.0 - 1.0 / sqrt(1.0 + t * t));
+	}
```

Comment:
This was removed intentionally. The purpose of vertex lighting is twofold:
1. To be very fast when you need fast but simple lighting (i.e for particles)
2. To allow users to make old school looking games. 

Neither of those cases require having area lights. so we dropped the code.  The nature of specular reflections in vertex shaded materials also makes the effect barely visible except at extreme values, so its really not worth the cost IMO. 

I wouldn't bring it back unless we have enough users saying that they absolutely need it.

Reply:
That's fair. I figured the cost wasn't much of a concern since it's only calculated when the user wants it to be.
That being said, if the _only_ concern is performance, I can easily optimize this so it only uses one quarter-rate instruction instead of three (or even none, as `size_A = max(0.0, 1.0 - 1.0 / sqrt(1.0 + t * t))` simply clamps the value below `1.0` with a very subtle rolloff, so can easily be dropped in favor of `size_A = clamp(t, 0.0, 1.0)`, which would bring the area light code down to a multiply, a clamp, and three adds (two if we also drop the geometry term I added)).

Reply:
I would wait and see if users ask for it. There is no point in throwing away performance for something that may never get used.

Reply:
Speaking as a user, this _is_ something that I want. Since area lights affect diffuse shading too, and our area lights aren't energy conserving (larger area lights simply emit more light than smaller ones), ignoring area lights causes a brightness discrepancy between per-vertex and per-pixel shaded materials. If I'm using per-vertex shading for particles, I want them to be lit consistently with the rest of their environment, regardless of my lighting setup.

Perhaps it's outside the scope of this discussion, but I don't think we should always wait for users to notice deficiencies before addressing them, especially when the performance cost for doing so can be reduced to as little as three instructions (two when not using specularity).